### PR TITLE
Minor fix - removed unnecessary toArrayObject cast

### PR DESCRIPTION
**Pull request by kisbg**
Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

File: src/runtime/GlobalObjectBuiltinArray.cpp [hunk 1]
```diff
@@ -674,13 +674,12 @@ static Value builtinArraySlice(ExecutionState& state, Value thisValue, size_t ar
     int64_t n = 0;
     // Let count be max(final - k, 0).
     // Let A be ArraySpeciesCreate(O, count).
-    Value val = arraySpeciesCreate(state, thisObject, std::max(((int64_t)finalEnd - (int64_t)k), (int64_t)0));
-    ArrayObject* array = val.asObject()->asArrayObject();
```
Comment (rerobika) [inline src/runtime/GlobalObjectBuiltinArray.cpp]: How about just simply removing the invalid `asArrayObject()` cast?

Comment (kisbg) [reply src/runtime/GlobalObjectBuiltinArray.cpp]: Valid point. Fixed it

**Comment by clover2123** [comment]
It seems that the return value of `ArraySpeciesCreate` can be an `Object` other than `ArrayObject` right? If so, would you check other usages of `ArraySpeciesCreate` which assume that the return type should be an `ArrayObject` too?

**Comment by kisbg** [comment]
@clover2123  Yes you are right. I removed similar to asArrayObject cast.

**Comment by ksh8281** [comment]
looks good
but could you change return type of arraySpeciesCreate as Object*?
that is more clear I think

**Comment by ksh8281** [comment]
@rerobika r?
if he accecpt, we can merge this pr

**Pull request merged**